core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1177,13 +1178,13 @@ public synchronized boolean trim() throws IOException {
             for (TransientFingerprintFacetFactory fff : TransientFingerprintFacetFactory.all()) {
                 fff.createFor(this,transientFacets);
             }
-            this.transientFacets = ImmutableList.copyOf(transientFacets);
+            this.transientFacets = Collections.unmodifiableList(transientFacets);
```

Comment:
The local variable `transientFacets` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.